Support client certificates for HTTP requests

sykaeh opened the pull request:
This fixes certtools/intelmq#737

I also used this opportunity to consolidate the parameters that are supplied for HTTP requests and added the parameter parsing in the general CollectorBot. 

There also seemed to be inconsistencies whether `http_ssl_proxy` or `https_proxy` should be used. I replaced all of them with `https_proxy` (seemed to be more common and documented).

sebix commented:
* Renaming parameters means breaking installations
* The newly added parameters are not documented in docs/Bots.md
* http_proxy and https_proxy are provided by defaults as None, you explicitly change it to empty string :(
* As they are always provided, no need to use the getattr workaround
* ssl_cl_cert is not self-explaining, better use ssl_client_cert

aaronkaplan commented:
overall : looks ok. Did a brief review of the code. Please let's test this internally and merge.
Thanks @sykaeh

sykaeh commented:
> Renaming parameters means breaking installations

I think for the sake of simplicity and clarity it is worth it. Also, if we change it before 1.0, we do not have to try to maintain the inconsistencies in the future. 

> The newly added parameters are not documented in docs/Bots.md

There are a lot of things that are not documented in docs/Bots.md. I have now added all of the generic collectors along with their parameters. As I have mentioned before in #757, it would be nice to have a better solution to this. Also, I do not understand what exactly is meant by `name`, `lookup` and `public` in the Bots.md file.

> http_proxy and https_proxy are provided by defaults as None, you explicitly change it to empty string :(

Sorry, I hadn't noticed that. I changed it to `null`.

> As they are always provided, no need to use the getattr workaround

I just copied what was done before, but I agree and have now changed it.

> ssl_cl_cert is not self-explaining, better use ssl_client_cert

Same, I just copied what was done before, but I changed it to ssl_client_cert
